theano/gpuarray/type.py

```diff
@@ -542,6 +517,65 @@ def c_code_cache_version(self):
         return (2, ver[0])
 
 
+def values_eq_approx(a, b, allow_remove_inf=False, allow_remove_nan=False,
+                     rtol=None, atol=None):
+    if a.shape != b.shape or a.dtype != b.dtype:
+        return False
+    if str(a.dtype) in theano.tensor.discrete_dtypes:
+        return GpuArrayType.values_eq(a, b)
+    else:
+        if allow_remove_inf or allow_remove_nan:
+            raise NotImplementedError(
+                "GpuArrayType.values_eq_approx() don't implemented the"
+                " allow_remove_inf and allow_remove_nan parameter")
```

Comment:
Maybe we only have to remove that error, and jump directly to the section after `# maybe the trouble is that there are NaNs`?

Reply:
Yes, I think that could work.